dali/test/python/test_operator_arithmetic_ops.py

```diff
@@ -186,89 +199,111 @@ def get_numpy_input(input, kind, type):
     if kind == "const":
         return type(magic_number)
     else:
-        return input.astype(type)
+        if "scalar" in kind:
+            return input.astype(type).reshape(input.shape + (1,))
+        else:
+            return input.astype(type)
 
 # Regular arithmetic ops that can be validated as straight numpy
-def check_arithm_op(kinds, types, op, _):
+def check_arithm_op(kinds, types, op, shape, _):
     left_type, right_type = types
     left_kind, right_kind = kinds
     target_type = bin_promote(left_type, right_type)
-    iterator = iter(ExternalInputIterator(batch_size, left_type, right_type))
+    iterator = iter(ExternalInputIterator(batch_size, shape, left_type, right_type, left_kind, right_kind))
     pipe = ExprOpPipeline(kinds, types, iterator, op, batch_size = batch_size, num_threads = 2,
             device_id = 0)
     pipe.build()
     pipe_out = pipe.run()
-    l = as_cpu(pipe_out[0]).as_array()
-    r = as_cpu(pipe_out[1]).as_array()
-    out = as_cpu(pipe_out[2]).as_array()
-    assert_equals(out.dtype, target_type)
-    l_np = get_numpy_input(l, left_kind, target_type)
-    r_np = get_numpy_input(r, right_kind, target_type)
-    if 'f' in np.dtype(target_type).kind:
-        np.testing.assert_allclose(out, op(l_np, r_np),
-            rtol=1e-07 if target_type != np.float16 else 0.005)
-    else:
-        np.testing.assert_array_equal(out, op(l_np, r_np))
+    for sample in range(batch_size):
+        l = as_cpu(pipe_out[0]).at(sample)
+        r = as_cpu(pipe_out[1]).at(sample)
```

Comment:
this code can be extracted to a function (you use it again below)

Reply:
done